x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BigArrayVectorTests.java

```diff
@@ -84,7 +84,10 @@ public void testBoolean() throws IOException {
                     assertThat(mask.mask().getBoolean(p), equalTo(values[p]));
                 }
             }
-            if (value != null) {
+            if (value == null) {
+                assertThat(vector.allTrue(), equalTo(Arrays.stream(values).allMatch(v -> v)));
+                assertThat(vector.allFalse(), equalTo(Arrays.stream(values).allMatch(v -> v == false)));
```

Comment:
It should be quite safe to just use this logic for all cases, and remove the if/else